solutions/standard/main.tf

```diff
@@ -19,12 +19,66 @@ locals {
   cos_kms_key_crn                  = var.existing_cos_bucket_name != null ? null : var.existing_kms_root_key_crn != null ? var.existing_kms_root_key_crn : module.kms[0].keys[format("%s.%s", local.cos_key_ring_name, local.cos_key_name)].crn
+
+  kms_service_name = var.existing_kms_instance_crn != null ? (
+    can(regex(".*kms.*", var.existing_kms_instance_crn)) ? "kms" : (
+      can(regex(".*hs-crypto.*", var.existing_kms_instance_crn)) ? "hs-crypto" : null
+    )
+  ) : null
+}
+
+# Data source to account settings for retrieving cross account id
+data "ibm_iam_account_settings" "iam_account_settings" {
+  count = local.create_cross_account_auth_policy ? 1 : 0
+}
+
+# Create IAM Authorization Policy to allow COS to access KMS for the encryption key
+resource "ibm_iam_authorization_policy" "cos_kms_policy" {
+  count = local.create_cross_account_auth_policy ? 1 : 0
+  # Conditionals with providers aren't possible, using ibm.kms as provider incase cross account is enabled
+  provider                    = ibm.kms
+  source_service_account      = data.ibm_iam_account_settings.iam_account_settings[0].account_id
+  source_service_name         = "cloud-object-storage"
+  source_resource_instance_id = local.cos_instance_guid
+  target_service_name         = local.kms_service_name
+  target_resource_instance_id = local.existing_kms_guid
+  roles                       = ["Reader"]
+  description                 = "Allow the COS instance with GUID ${local.cos_instance_guid}  to read from the ${local.kms_service_name} instance GUID ${local.existing_kms_guid}"
+}
+
+# Create IAM Authorization Policy to allow EN to access KMS for the encryption key
+resource "ibm_iam_authorization_policy" "en_kms_policy" {
+  count = local.create_cross_account_auth_policy ? 1 : 0
+  # Conditionals with providers aren't possible, using ibm.kms as provider incase cross account is enabled
+  provider                    = ibm.kms
+  source_service_account      = data.ibm_iam_account_settings.iam_account_settings[0].account_id
+  source_service_name         = "event-notifications"
+  source_resource_instance_id = module.event_notifications.guid
+  target_service_name         = local.kms_service_name
+  target_resource_instance_id = local.existing_kms_guid
+  roles                       = ["Reader"]
+  description                 = "Allow the EN instance with GUID ${module.event_notifications.guid} reader access to the kms_service_name instance GUID ${local.existing_kms_guid}"
+
+}
+
+# workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
+resource "time_sleep" "wait_for_authorization_policy_en" {
+  depends_on      = [ibm_iam_authorization_policy.en_kms_policy]
+  create_duration = "30s"
+}
+
+# workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
+resource "time_sleep" "wait_for_authorization_policy_cos" {
+  depends_on      = [ibm_iam_authorization_policy.cos_kms_policy]
+  create_duration = "30s"
```

Comment:
this time sleep is pointless - your not using it as part of any depends_on. I also don't think you can use it anywhere because ideally the sleep would need to occur after the COS instance is created, but before the bucket is created, and since both instance and bucket are created as part of the COS module, there is nothing we can do but reply on the terraform dependency tree to do its job